### DISCUSSION TITLE
How do you configure your dashboard now?

### DISCUSSION THREAD

**Discussion by voodookiidoo** [LazyVim/LazyVim#4865] [Q&A] [answered]
Just a week ago i figured out a way to configure my dashboard and was so happy. And now that snacks came - i have no idea how to make snakcs load up my dashboard, and no the default one.
Previously it was defined like this
`  {
    "nvimdev/dashboard-nvim",
    lazy = false,
    dependencies = {
      { "nvim-tree/nvim-web-devicons" },
    },
    config = function()
      require("voodoo.dashboard")
    end,
  },
`

but now i don't know how to pass a custom config inside snacks to make it load my own dashboard
`  {
    "folke/snacks.nvim",
    enabled = true,
    config = function()
      require("voodoo.snackboard")
    end,

    opts = {
      dashboard = {

        enabled = true,
      },
    },
  },
`
Do you pass it inside dashboard options?
Snacks options?
Neither of it works. Please help :(

**Comment by folke** [comment, marked as answer]
Just install the dashboard-nvim extra to keep your old dashboard.
Or read the snakcs dashboard docs to use the new one.

**Reply by voodookiidoo** [reply]
Damn that was fast!
Thanks man
But really - i like your new version of dashboard, but i can't find any info how to use it (i mean apply Snacks.dashboard config to it

**Reply by folke** [reply]
first of all, don't use config, but opts instead

```lua
{
  "folke/snacks.nvim",
  opts = {
    dashboard = {...}
  }
} 
```

And here's the docs https://github.com/folke/snacks.nvim/blob/main/docs/dashboard.md

**Reply by voodookiidoo** [reply]
Yeah, i saw that, but i can't find an example of how to connect my new plugins config to the lazy config. Do you specify it as?
`
{
  "folke/snacks.nvim",
  opts = {
    dashboard = {config = function()
    requiere("voodooo.snackboard")
    end,
    }
  }
} 
`
I want an example of connecting my config to the lazy config, not the config itself (i already saw that)

**Reply by sebiwi** [reply]
Hello @folke , I think I'm trying to do what you said here:

```
-- nvim/lua/plugins/extend-dashboard.lua
return {
  "folke/snacks.nvim",
  opts = {
    dashboard = {
      sections = {
        { section = "header" },
        { section = "keys", gap = 1, padding = 1 },
        {
          icon = " ",
          title = "Git Status",
          section = "terminal",
          enabled = Snacks.git.get_root() ~= nil,
          cmd = "hub status --short --branch --renames",
          height = 5,
          padding = 1,
          ttl = 5 * 60,
          indent = 3,
        },
        { section = "startup" },
      },
    },
  },
}
```

I get the following error on startup:

```
   Error  10:56:28 notify.error lazy.nvim Failed to load `plugins.extend-dashboard`

/Users/home/.config/nvim/lua/plugins/extend-dashboard.lua:24: attempt to index global 'Snacks' (a nil value)

# stacktrace:
  - ~/.config/nvim/lua/plugins/extend-dashboard.lua:24 _in_ **load**
  - ~/.config/nvim/lua/config/lazy.lua:17
  - nvim/init.lua:2
   Warn  11:01:02 notify.warn lazy.nvim # Config Change Detected. Reloading...

- **changed**: `~/.config/nvim/lua/plugins/extend-dashboard.lua`
   Warn  11:02:16 notify.warn lazy.nvim # Config Change Detected. Reloading...

- **changed**: `~/.config/nvim/lua/plugins/extend-dashboard.lua`
10:56:39 msg_showcmd gk

```

I'm using the starter template. Am I doing something wrong?

Thanks!

**Reply by dpetka2001** [reply]
Try 
```lua
enabled = function()
  return Snacks.git.get_root() ~= nil
end,
```

**Reply by sebiwi** [reply]
@dpetka2001 thanks for the feedback. I tried adding that inside of `ops`, like this:

```
return {
  "folke/snacks.nvim",
  opts = {
    enabled = function()
      return Snacks.git.get_root() ~= nil
    end,
    dashboard = {
      sections = {
        { section = "header" },
        { section = "keys", gap = 1, padding = 1 },
        {
          icon = " ",
          title = "Git Status",
          section = "terminal",
          enabled = Snacks.git.get_root() ~= nil,
          cmd = "hub status --short --branch --renames",
          height = 5,
          padding = 1,
          ttl = 5 * 60,
          indent = 3,
        },
        { section = "startup" },
      },
    },
  },
}
```

And also inside of `dashboard`, like this:


```
return {
  "folke/snacks.nvim",
  opts = {
    dashboard = {
      enabled = function()
        return Snacks.git.get_root() ~= nil
      end,
      sections = {
        { section = "header" },
        { section = "keys", gap = 1, padding = 1 },
        {
          icon = " ",
          title = "Git Status",
          section = "terminal",
          enabled = Snacks.git.get_root() ~= nil,
          cmd = "hub status --short --branch --renames",
          height = 5,
          padding = 1,
          ttl = 5 * 60,
          indent = 3,
        },
        { section = "startup" },
      },
    },
  },
}
```

I still get the same error when opening nvim:

```
   Error  12:34:19 notify.error lazy.nvim Failed to load `plugins.extend-dashboard`

/Users/home/.config/nvim/lua/plugins/extend-dashboard.lua:17: attempt to index global 'Snacks' (a nil value)

# stacktrace:
  - ~/.config/nvim/lua/plugins/extend-dashboard.lua:17 _in_ **load**
  - ~/.config/nvim/lua/config/lazy.lua:17
  - nvim/init.lua:2
```

**Reply by dpetka2001** [reply]
You have to put it in the section of git status that you use. So here
```lua
{
          icon = " ",
          title = "Git Status",
          section = "terminal",
          enabled = Snacks.git.get_root() ~= nil,
          cmd = "hub status --short --branch --renames",
          height = 5,
          padding = 1,
          ttl = 5 * 60,
          indent = 3,
        },
```
just replace the `enabled` that you have with what I mentioned.

**Reply by sebiwi** [reply]
That worked, thanks! Is `enabled` different from what's defined [here](https://github.com/folke/snacks.nvim/blob/main/docs/dashboard.md#advanced) because of LazyVim?

**Reply by dpetka2001** [reply]
It seems that `Snacks` global is not available during the parsing phase of files from `lazy.nvim`. That's why you wrap it into a function, so that it gets evaluated when it's actually called when the dashboard loads. If you just have it as key/value inside a table, then it gets evaluated during the parsing phase of `lazy.nvim` (the package manager) and that's why you get the error. It has to do with when `lazy.nvim` (the package manager) evaluates the values during it's parsing phase.
